src/net/discovery/mod.rs

```diff
@@ -12,28 +12,52 @@ pub enum Error {
 }
 
 pub mod peer {
-    use axum::http::Uri;
+    use hyper::Uri;
+    #[cfg(feature = "enable-serde")]
+    use serde::de::Error;
+    #[cfg(feature = "enable-serde")]
+    use serde::{Deserialize, Deserializer};
     use x25519_dalek::PublicKey;
 
-    #[derive(Clone)]
+    #[derive(Clone, Debug)]
+    #[cfg_attr(feature = "enable-serde", derive(serde::Deserialize))]
     pub struct HttpConfig {
-        pub origin: Uri,
+        #[cfg_attr(feature = "enable-serde", serde(deserialize_with = "pk_from_str"))]
         pub public_key: PublicKey,
     }
 
-    #[derive(Clone)]
+    #[derive(Clone, Debug)]
+    #[cfg_attr(feature = "enable-serde", derive(serde::Deserialize))]
     pub struct Config {
-        pub http: HttpConfig,
+        #[cfg_attr(feature = "enable-serde", serde(deserialize_with = "uri_from_str"))]
+        pub origin: Uri,
```

Comment:
so how does a helper know which endpoint itself is?

also, should this be represented as a map of `origin -> conf`? otherwise, when we receive mapping `secret_shares -> origin` from report collector, i guess we would have to iterate through array?

Reply:
I was thinking that in prod each helper would have a single configuration file where the specific helper configuration is written. So there will be no ambiguity which endpoint to pick and listen to.

for test case, it does not really matter as long as each helper picks a unique entry from the config file. so incrementing an index and returning config[index] would probably work just fine?

Reply:
gotcha. so we have an as-of-yet undefined config that would include specific helper config.